src/macros.cr

```diff
@@ -46,6 +46,16 @@
 # Point.new      # => #<Point(@x=0, @y=0)>
 # Point.new y: 2 # => #<Point(@x=0, @y=2)>
 # ```
+#
+# This macro also provides a `copy_with` method which returns
+# a copy of the record with the provided properties altered.
+#
+# ```
+# record Point, x = 0, y = 0
+#
+# p = Point.new y: 2 # => #<Point(@x=0, @y=2)>
+# p.copy_with x: 3   # => #<Point(@x=3, @y=2)>
+# ```
```

Comment:
The sample could show that p is not changed
```crystal
point = Point.new y: 2 # => #<Point(@x=0, @y=2)>
point.copy_with x: 3   # => #<Point(@x=3, @y=2)>
point                  # => #<Point(@x=0, @y=2)>
```

Reply:
Fixed